### PR TITLE
docs(mssql): clarify in comments and docs supports MS SQL versions

### LINKED ISSUE #13838: Clarify supported Microsoft SQL Server versions
## Issue Creation Checklist

- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

In `.github/workflows/ci.yml:200` there is the line
```
mssql-version: [2017, 2019]
```

In `lib/dialects/mssql/index.js:63` there is the line
```
MssqlDialect.prototype.defaultVersion = '12.0.2000'; // SQL Server 2014 Express, minimum supported version
```
In the documentation, there isn't a referenced minimum version
https://sequelize.org/master/manual/dialect-specific-things.html#mssql

If you need to refresh which versions of MS SQL Server exist, there's wikipedia, which also includes their end-of-support dates.
https://en.wikipedia.org/wiki/SQL_Server_Express#Version_history

### Possible Solution

The issue is that the three sources above don't match, and hence it is not entirely clear which is the minimum version supported.
If 2014 (version 12) shall be supported, there should at the very least be a CI job testing against that version. Microsoft is quite good with backwards compatibility so testing for that version should ensure compatibility with all newer versions.

Else, `mssql/index.js`  should be updated to mention version 14 and the version should be added to the docs.

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [x] Yes. This issue only applies to the following dialect(s): MS SQL
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by joaoe**
### Pull Request Checklist

_Please make sure to review and check all of these items:_

- [ ] Have you added new tests to prevent regressions?
- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Closes #13838

### CODE REVIEW

**Comment by joaoe** [comment]
@WikiRik There you go

File: docs/manual/other-topics/upgrade-to-v7.md [hunk 1]
```diff
@@ -30,3 +30,10 @@ sequelize.connectionManager.getConnection({ type: 'SELECT' });
 // Do this:
 sequelize.connectionManager.getConnection({ type: 'read' });
 ```
+
```
Comment (WikiRik) [inline docs/manual/other-topics/upgrade-to-v7.md:33]: You are only allowed to have 1 blank line here

File: docs/manual/other-topics/upgrade-to-v7.md [hunk 2]
```diff
@@ -30,3 +30,10 @@ sequelize.connectionManager.getConnection({ type: 'SELECT' });
...
+
+### Microsoft SQL Server Support
+
+Sequelize v7 fully supports MS SQL Server 2017 (version 14) onwards, up from 2012 (version 13) in
+Sequelize v6, as this matches Microsoft's own [mainstream support](
+https://en.wikipedia.org/wiki/SQL_Server_Express#Version_history).
```
Comment (WikiRik) [inline docs/manual/other-topics/upgrade-to-v7.md:39]: Why didn't you use the official site?; https://docs.microsoft.com/en-us/sql/sql-server/end-of-support/sql-server-end-of-life-overview?view=sql-server-ver15#lifecycle-dates
And maybe add somewhere that earlier versions might be supported but are not tested

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -207,7 +207,7 @@ jobs:
       fail-fast: false
       matrix:
         node-version: [12, 16]
-        mssql-version: [2017, 2019]
+        mssql-version: [2017, 2019]  # See lib/dialects/mssql/index.js
```
Comment (WikiRik) [inline .github/workflows/ci.yml:210]: Not sure if we actually need this comment here but can't hurt

**Review by WikiRik — COMMENTED** [review]
Some small comments. Thanks for making a separate PR for this!

**Comment by joaoe** [comment]
@WikiRik All fixed. Waiting for you :) Thanks

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
Looking good, thanks again!

**Comment by joaoe** [comment]
Great, cheers !

**Pull request merged**